ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryExecutor.java

```diff
@@ -209,6 +234,12 @@ PersistentQueryMetadata buildPersistentQuery(
 
     final RuntimeBuildContext runtimeBuildContext = buildContext(applicationId, queryId);
     final Object result = buildQueryImplementation(physicalPlan, runtimeBuildContext);
+    // Must be created before the topology is created since the ScalablePushRegistry adds a peek
+    // to the topology.
```

Comment:
this goes back to my other point about it being weird to register the peek call within the constructor

Reply:
I agree.  Made the suggested change.  I still added some comments here to make it clear what's going on, but I changed the method to `applyScalablePushProcessor` so it should be clearer.